lib/ansible/module_utils/vmware.py

```diff
@@ -482,10 +484,33 @@ def vmware_argument_spec():
         validate_certs=dict(type='bool',
                             required=False,
                             default=True,
-                            fallback=(env_fallback, ['VMWARE_VALIDATE_CERTS'])),
+                            fallback=(env_fallback, ['VMWARE_VALIDATE_CERTS'])
+                            ),
+        https_proxy=dict(type='str',
```

Comment:
I think we should just expose a `proxy_host` and and `proxy_port` parameters. This way we avoid all the complexity of the URL parsing and we don't need to deal with the environment variables (https_proxy/http_proxy/HTTPS_PROXY/no_proxy/etc).